minpy/primitive.py

```diff
@@ -153,7 +192,7 @@ def def_grad(self, func, argnum=0):
         :param argnum: Index of the argument.
         :return: Self.
         """
-        self._grad_func[argnum] = func
+        self._grad_func[argnum] = GradFunc(f=func, multi_grad_indices=None)
         return self
 
     def def_grad_kw(self, func, key):
```

Comment:
My idea is to redirect `def_grad` and `def_grad_kw` all to `def_multi_grad`. They are just special cases.